src/main/java/org/mifos/connector/mpesa/utility/SafaricomUtils.java

```diff
@@ -83,19 +81,35 @@ public Long getTimestamp() {
         return Long.parseLong(sdf.format(new Date()));
     }
 
+    /*
+     * Return the result code from the callback received from mpesa server
+     */
+    public static StkCallback getResultCode(JsonObject callback) {
+        StkCallback stkCallback = new StkCallback();
+
+        LinkedHashMap<String, Object> body = (LinkedHashMap<String, Object>) callback.get("Body");
+        LinkedHashMap<String, Object> stkCallbackFromServer = (LinkedHashMap<String, Object>) body.get("stkCallback");
+        stkCallback.setMerchantRequestId((String) stkCallbackFromServer.get("MerchantRequestID"));
+        stkCallback.setCheckoutRequestId((String) stkCallbackFromServer.get("CheckoutRequestID"));
+        stkCallback.setResultCode(Long.parseLong(String.valueOf(stkCallbackFromServer.get("ResultCode"))));
+        stkCallback.setResultDesc((String) stkCallbackFromServer.get("ResultDesc"));
+
+        return stkCallback;
+    }
+
     /*
      * Return the transaction id from the callback received from mpesa server
      */
     public static String getTransactionId(JsonObject callback) {
         AtomicReference<String> mpesaReceiptNumber = new AtomicReference<>("");
-        JsonObject body = (JsonObject) callback.get("Body");
-        JsonObject metaData = (JsonObject) body.get("CallbackMetadata");
-        JsonArray metaDataItems = (JsonArray) metaData.get("items");
+        LinkedHashMap<String, Object> body = (LinkedHashMap<String, Object>) callback.get("Body");
+        LinkedHashMap<String, Object> metaData = (LinkedHashMap<String, Object>) body.get("CallbackMetadata");
+        ArrayList<Object> metaDataItems = (ArrayList) metaData.get("Item");
```

Comment:
If you're only going to use metaDataItems, perhaps we can simply chain the access instead of defining multiple intermediate variables.

Reply:
I made it this way, so that code looks more clear. Should I chain the statement?

Reply:
![Screenshot 2022-01-19 at 6 17 04 PM](https://user-images.githubusercontent.com/31315800/150133550-71f09cac-3085-46ee-9c3e-8a8ffca5f9b8.png)

It will look something like this, I think it will be complicated for later references. Whatever you suggest, should I chain it?